taichi/runtime/program_impls/vulkan/vulkan_program.h

```diff
@@ -47,7 +47,9 @@ class VulkanProgramImpl : public ProgramImpl {
   void materialize_snode_tree(SNodeTree *tree, uint64 *result_buffer) override;
 
   void synchronize() override {
-    vulkan_runtime_->synchronize();
+    if (vulkan_runtime_) {
```

Comment:
When is `vulkan_runtime_` not inited?

Reply:
@k-ye when you have created a program for compilation only (not execution) e.g. https://github.com/taichi-dev/taichi/blob/master/tests/cpp/aot/aot_save_load_test.cpp#L23-L108 
(ofc it works since these kernels didn't require jit evaluation.

Reply:
... sounds brittle